app/services/post/UpdateService.php

```diff
@@ -47,8 +51,49 @@ public function execute(Request $request, string $id)
                 PostToTag::create($pdo, $id, $tag_id);
             }
 
+            $current_post_image_urls = [];
+            if (isset($request->files['images'])) {
+                // 現在の画像のパスをデータ削除前に取得しておき、トランザクションのコミット後に削除を試行
+                // (トランザクションをロールバックした場合、画像削除はロールバックできないため)
+                $current_post_image_urls = array_map(fn ($post_image) => $post_image->image_url, PostImage::getAllByPostId($pdo, $id));
+                
+                // サムネイルを NULL に更新
+                $post->updateThumbnailPostImageId($pdo, null);
+                // DB から PostImage を削除
+                PostImage::bulkDestroyByPostId($pdo, $id);
+
+                // 画像の作成
+                $uploaded_images = $request->files['images'];
+                // アップロードされた枚数分繰り返し
+                // TODO: O(n) の回避
+                for ($index = 0; $index < count($uploaded_images['tmp_name']); $index ++) {
+                    $file = array(
+                        'name' => $uploaded_images['name'][$index],
+                        'tmp_name' => $uploaded_images['tmp_name'][$index],
+                        'size' => $uploaded_images['size'][$index],
+                    );
```

Comment:
```suggestion
                    $file = [
                        'name' => $uploaded_images['name'][$index],
                        'tmp_name' => $uploaded_images['tmp_name'][$index],
                        'size' => $uploaded_images['size'][$index],
                    ];
```
のような短縮構文のほうが他言語と近い書き方なので直感的な気はします。

Reply:
@SardineTa23 
なるほど！
連想配列の時は `array()` なのかと文法を勝手に勘違いしていました。ありがとうございます:pray:

Reply:
@you-5805 それでいうとPHPに多言語の配列は存在せず、HashとかMapといわれるやつを配列っぽく使っている感じですね。便利！！！！（？）